### PR TITLE
Detach integration tests to a separate module

### PR DESCRIPTION
**Pull request by ndeloof**
see #282

### CODE REVIEW

**Comment by ndeloof** [comment]
I haven't moved `SeedJobTest` to `integration` as I get a weird failure with missing workflow-multibranch plugin. Will investigate on this later.

File: pom.xml [hunk 1]
```diff
@@ -20,7 +24,7 @@
     <java.level>8</java.level>
   </properties>
 
-  <name>Configuration as Code Plugin</name>
+  <name>Configuration as Code</name>
```
Comment (oleg-nenashev) [inline pom.xml]: "... Parent POM"

File: pom.xml [hunk 2]
```diff
@@ -1,17 +1,21 @@
...
-
   <parent>
     <groupId>org.jenkins-ci.plugins</groupId>
     <artifactId>plugin</artifactId>
     <version>3.10</version>
     <relativePath />
   </parent>
-  <groupId>io.jenkins</groupId>
-  <artifactId>configuration-as-code</artifactId>
+  <groupId>io.jenkins.configuration-as-code</groupId>
```
Comment (oleg-nenashev) [inline pom.xml]: maybe we should also add "plugins" to groupId

Comment (ndeloof) [reply pom.xml]: use of io.jenkins was by intent, as we expected this project to become a top level one (like remoting is).

**Pull request merged**